DonjinKrawler-Client/src/main/java/donjinkrawler/flyweight/EnemyShellForTest.java

```diff
@@ -0,0 +1,112 @@
+package donjinkrawler.flyweight;
+
+import donjinkrawler.AbstractShell;
+import donjinkrawler.Game;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class EnemyShellForTest extends AbstractShell {
+
+    private int id;
+    private String info = "No strategy";
+    private Map<String, ImageIcon> clothes = new HashMap<>();
+    private EnemyType type;
+
+    public EnemyShellForTest(String name, int id, int x, int y) {
+        this.name = name;
+        this.id = id;
+        this.x = x;
+        this.y = y;
+    }
+
+    @Override
+    public void loadImage() {
+        ImageIcon ii = new ImageIcon(ClassLoader.getSystemResource(this.resolveImageName()).getFile());
+        image = ii.getImage();
+    }
+
+    public Image getImage() {
+        return image;
+    }
+    private String resolveImageName() {
```

Comment:
Maybe extract this to a static helper? It's a duplicate in EnemyType, would be easier to maintain if we were to add more enemies.

Reply:
I don't think any optimization is required for EnemyShellForTest because the class is only created to show the use of Flyweight implementation and is not actually required to run the game